src/main/kotlin/org/elm/ide/actions/ElmExternalFormatAction.kt

```diff
@@ -27,14 +29,10 @@ class ElmExternalFormatAction : AnAction() {
     private fun getContext(e: AnActionEvent): Context? {
         val project = e.project ?: return null
         val toolchain = project.elmToolchain ?: return null
-        val file = e.getData(CommonDataKeys.PSI_FILE) ?: return null
+        val file: PsiFile = e.getData(CommonDataKeys.PSI_FILE) ?: return null
         if (!file.virtualFile.isInLocalFileSystem) return null
         if (file !is ElmFile) return null
-        val elmVersion = when (val elmProject = file.elmProject) {
-            is ElmApplicationProject -> elmProject.elmVersion
-            is ElmPackageProject -> elmProject.elmVersion.low
-            else -> return null
-        }
+        val elmVersion = getElmVersion(file) ?: return null
```

Comment:
Your change no longer uses the `elmVersion` from the `Context` object. So you can remove this line entirely. That will leave `getElmVersion()` with only a single caller, and it should probably be inlined.